ruuvi_examples/ruuvi_firmware/main.c

```diff
@@ -282,97 +296,149 @@ ret_code_t lis2dh12_int2_handler(const ruuvi_standard_message_t message)
 }
 
 
-/**
- * @brief Function for application main entry.
- */
+/**  This is where it all starts ++++++++++++++++++++++++++++++++++++++++++ 
+ main is entered as a result of one of SEVERAL events:
+  - Normal startup from press of reset button.
+  - Battery inserted.
+  - After DFU (Device Firmware Upgrade) at manufacturing Quality Assurance or user DFU.
+  - WatchDogTimer expiration and its interrupt handler didn't feed new value.
+  - Some error occured and
+  - Spontenous unknown reset.
+ All subsystems are initalized and any failures are noted and available later in init_status
+ Since some events occur after tag is deployed and no one can see the LEDs the system continues operating.
+
+ After initalizition (including setting up interrupts)
+    we loop here calling app_sched_execute and sd_app_evt_wait 
+*/
 int main(void)
 {
-  ret_code_t err_code = NRF_SUCCESS; // Error flag, OR any errors to it and check the error code at the end.
+ init_leds();                      // LEDs first (they're easy and cannot fail)  drivers/init/init.c ( RED got set up earlier)
+ nrf_delay_ms(1000u);              // sit here for a second with LEDs off

```

Comment:
What is the reasoning for waiting with the LEDs off? Please note that the bootloader starts with red LED on, and therefore this gives behavior RED-OFF_RED

Reply:
The delay permits the viewing of the version blinks distinctly. Reducing the RED-OFF _RED observation.
Would you like to see what it looks like? I can prepare a video.

Reply:
Removed